### PR TITLE
Feature/check max core number

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### This PR fixes:

- Moved sub check_max_core_number

### How to test:

- Automatic and continuous test pass

### Expected outcome:
- Installation, unit and integration tests pass

### Review:
- [ ] Code review
- [ ] New code is executed and covered by tests
- [ ] Tests pass

### CODE REVIEW

File: lib/MIP/Environment/Cluster.pm [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    ## Check number of cores available
+    if ( $core_number_requested > $core_constraint ) {
+
+        ## Return maximum available cores
+        return $core_constraint;
+    }
+
+        ## Core number requested is lower than the number of available cores
```
Comment (jemten) [inline lib/MIP/Environment/Cluster.pm]:
```suggestion
    ## Core number requested is lower than the number of available cores
```

File: lib/MIP/Environment/Cluster.pm [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        return $core_number_requested;
```
Comment (jemten) [inline lib/MIP/Environment/Cluster.pm]:
```suggestion
    return $core_number_requested;
```

File: lib/MIP/Main/Analyse.pm [hunk 3]
```diff
@@ -359,18 +360,8 @@ sub mip_analyse {
...
-        $active_parameter{recipe_core_number}{$recipe_name} = check_max_core_number(
-            {
-                max_cores_per_node => $active_parameter{max_cores_per_node},
-                core_number_requested =>
-                  $active_parameter{recipe_core_number}{$recipe_name},
-            }
-        );
-    }
+    ## Check core number requested against environment provisioned
+    parse_recipe_resources( { active_parameter_href => \%active_parameter, } );
```
Comment (jemten) [inline lib/MIP/Main/Analyse.pm]: much prettier!

Comment (henrikstranneheim) [reply lib/MIP/Main/Analyse.pm]: Thx!

**Pull request merged**